### ISSUE TITLE
Error reported attempting to open resources found in Discover

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#4325]
**Description of the bug**
Attempts to open specific resources found in Discover result in the following error 

Steps to reproduce the bug:
1. Go to hydroshare.org/search
2. enter 'garousi' in the search bar and enter
3. Click on the second resource on the list with title "Notebooks for post-processing ..." 

![image](https://user-images.githubusercontent.com/3456833/132603162-f75f4a77-e50a-4799-b901-30941584119a.png)

**Expected behavior**
The resource landing page should open

The URL that attempts to open is https://www.hydroshare.org/resource/1b66a752b0cc467eb0f46bda5fdc4b34/.

**Comment by sblack-usu** [comment]
Here is the error generated in the logs.

```
[09/Sep/2021 15:59:15] ERROR [django.request:135] Internal Server Error: /resource/1b66a752b0cc467eb0f46bda5fdc4b34/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 882, in _resolve_lookup
    current = current[bit]
TypeError: 'CompositeResource' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 217, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 215, in _get_response
    response = response.render()
  File "/usr/local/lib/python3.6/site-packages/django/template/response.py", line 107, in render
    self.content = self.rendered_content
  File "/usr/local/lib/python3.6/site-packages/django/template/response.py", line 84, in rendered_content
    content = template.render(context, self._request)
  File "/usr/local/lib/python3.6/site-packages/django/template/backends/django.py", line 66, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 207, in render
    return self._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/test/utils.py", line 107, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/test/utils.py", line 107, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/test/utils.py", line 107, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/test/utils.py", line 107, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 72, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/defaulttags.py", line 322, in render
    return nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 1040, in render
    output = self.filter_expression.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 708, in resolve
    obj = self.var.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 849, in resolve
    value = self._resolve_lookup(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 890, in _resolve_lookup
    current = getattr(current, bit)
  File "/hydroshare/hs_core/models.py", line 3611, in size
    f.calculate_size()
  File "/hydroshare/hs_core/models.py", line 3007, in calculate_size
    self._size = self.resource_file.size
  File "/usr/local/lib/python3.6/site-packages/django/db/models/fields/files.py", line 77, in size
    return self.storage.size(self.name)
  File "/hydroshare/django_irods/storage.py", line 400, in size
    return int(float(stdout))
ValueError: could not convert string to float: '0\n1297064500\n'
```

**Comment by sblack-usu** [comment]
The error occurs because python is trying to convert `0\n1297064500\n` to a float.  That string should be the file size from  irods commands.  We need to figure out why irods is giving us that a value with newlines and how to handle or correct it.

**Comment by sblack-usu** [comment]
I am including more information from the owner of the resource.  I looked at the database and noticed there were two ResourceFile entries for `1b66a752b0cc467eb0f46bda5fdc4b34/data/contents/NWM_FSNO.csv` in the resource.  One of them correctly created the file size, the other did not.  This bad database state likely occurred when the files were moved.  I went ahead and removed the duplicate ResourceFile entry and the resource works again.  I'll bring this issue to the Monday Dev call for further discussion around potential problems when moving files.

> From: Irene Garousi-Nejad <i.garousi@aggiemail.usu.edu> 
Sent: Thursday, September 9, 2021 9:09 AM
To: David Tarboton <david.tarboton@usu.edu>
Subject: Re: Spelling
>
>This was working fine the last time I checked the resources before submitting our paper revision. This resources contained csv files greater than 1 GB, and I attempted to move around the csv files on the HydroShare webpage. Could that be the reason? We can discuss this issue in our meeting today.
>
>Thanks,
Irene
>
>On Thu, Sep 9, 2021 at 5:00 AM David Tarboton <david.tarboton@usu.edu> wrote:
Thanks.  See also https://github.com/hydroshare/hydroshare/issues/4325.  While answering the question from David Arctur that I cc’d you on I found a resource that seems broken.  It is one of the resources in the same collection.   I will be following up with the development team, but wanted to see if you knew anything odd about this one or if there were problems you had had with it.
> 
>Dave
